src/model.jl

```diff
@@ -1,106 +1,275 @@
-function BatchNormWrap(out_ch)
-    Chain(x->expand_dims(x,2),
-	  BatchNorm(out_ch),
-	  x->squeeze(x))
-end
+using Flux
+using Flux: @functor
 
-UNetConvBlock(in_chs, out_chs, kernel = (3, 3)) =
-    Chain(Conv(kernel, in_chs=>out_chs,pad = (1, 1);init=_random_normal),
-	BatchNormWrap(out_chs),
-	x->leakyrelu.(x,0.2f0))
+struct ConvBlock
+  op
+end
 
-ConvDown(in_chs,out_chs,kernel = (4,4)) =
-  Chain(Conv(kernel,in_chs=>out_chs,pad=(1,1),stride=(2,2);init=_random_normal),
-	BatchNormWrap(out_chs),
-	x->leakyrelu.(x,0.2f0))
+@functor ConvBlock
 
-struct UNetUpBlock
-  upsample
+function ConvBlock(in_channels, out_channels, kernel_sizes = [(3,3), (3,3)];
```

Comment:
We can mirror the Flux `Conv` API here - this would mean we switch out the positions of the kernels, and make the channels as a `Pair`. This does lose out on having default kernel pairs though. Are we expecting to not need to define the kernels much? I suppose we wouldn't need to, but maintaining that consistency in the API would be a plus. What do you think?

Reply:
Yes, good idea. There is a way to allow optional first arguments, since the number of arguments changes. Therefore we made the first argument now optional, followed by the mentioned `Pair`.